### PR TITLE
UofT-DSI | LCR - Assignment 1

### PR DESCRIPTION
**Pull request by Chun-YuanChen**
## What changes are you trying to make? (e.g. Adding or removing code, refactoring existing code, adding reports)
I built and tested a KNN model using the Wine dataset from the scikit-learn package. It achieved very good accuracy. I also added a few notes for communication and my thoughts.

## What did you learn from the changes you have made?
Through this practice, it gradually became clearer to me how to work through a basic ML model (i.e., from splitting the data, initializing a classifier, conducting a grid search for the hyperparameter, finding its optimal value, to evaluating the KNN model).

## Was there another approach you were thinking about making? If so, what approach(es) were you thinking of?
This task focuses on a single hyperparameter (the number of neighbors) and uses grid search. I think it would be beneficial to try tuning additional hyperparameters and experimenting with other search methods to compare their effects on model performance. It could also be helpful to bring in more performance metrics for discussion.

## Were there any challenges? If so, what issue(s) did you face? How did you overcome it?
KNN involves many steps and layers (like loops), and I think we’ll encounter the same with other ML/DL methods. It is easy to get lost if you don’t pay close attention to each line of code or really understand what they are for. To overcome these, I find it helpful to go through additional relevant articles and materials.

## How were these changes tested?
I checked detailed information more than just the numbers generated from the code. For example, instead of just relying on print(df_train_tune.best_params_), I also looked at the full accuracies_grid. And for the accuracy score, I conducted two different ways to cross-check and they both stood on the same page.

## A reference to a related issue in your repository (if applicable)
None.

## Checklist
- [x] I can confirm that my changes are working as intended

### CODE REVIEW

**Comment by efantinatti** [comment]
Hi Chun-Yuan  

# PR Feedback Report

**Student**: Chun-Yuan  
**GitHub Username**: @Chun-YuanChen 
**PR Number**: 1  
**Branch**: assignment-1  
**Status**: COMPLETE  
**PR URL**: https://github.com/Chun-YuanChen/LCR/pull/1

## What You Did Well (Hits)

Question 1: Question 1 (i): Correctly identified 178 observations.; Question 1 (ii): Correctly identified 14 variables..
Question 2: Question 2: Standardization and data splitting code found..
Question 3: Question 3: Grid search with KNeighborsClassifier, correct parameter grid, and best n_neighbors=8 identified..


## Areas for Improvement (Errors)

None.


## Overall Assessment

All good.
